src/Swift.Bindings/src/Emitter/StringEmitter/Handler/Method/MethodHandler.cs

```diff
@@ -44,42 +43,99 @@ public MethodHandler()
+            SignatureHandler signatureHandler = new SignatureHandler(methodEnv);
+            Signature pinvokeSignature = signatureHandler.GetPInvokeSignature();
+            Signature wrapperSignature = signatureHandler.GetWrapperSignature();
+            PInvokeBuilder pInvokeBuilder = methodEnv.PInvokeBuilder;
+            WrapperBuilder wrapperBuilder = methodEnv.WrapperBuilder;
 
-            if (methodEnv.MethodDecl.GenericParameters.Any(x => x.Constraints.Count > 0))
-            {
-                Console.WriteLine($"Method {methodEnv.MethodDecl.Name} has unsupported generic constraints");
-                return;
-            }
+            var staticKeyword = methodEnv.MethodDecl.MethodType == MethodType.Static || methodEnv.ParentDecl is ModuleDecl ? "static " : "";
+            var unsafeKeyword = methodEnv.RequiresIndirectResult || methodEnv.MethodDecl.IsGeneric ? "unsafe " : "";
+            string[] modifiers = new string[] { staticKeyword, unsafeKeyword }.Where(m => !string.IsNullOrEmpty(m)).ToArray();
 
-            if (signatureHandler.GetWrapperSignature().ContainsPlaceholder)
-            {
-                Console.WriteLine($"Method {methodEnv.MethodDecl.Name} has unsupported signature: ({signatureHandler.GetWrapperSignature().ParametersString()}) -> {signatureHandler.GetWrapperSignature().ReturnType}");
-                return;
-            }
+            string[] generics = methodEnv.MethodDecl.GenericParameters.Select(p => methodEnv.GenericTypeMapping[p.TypeName].PlaceholderName.ToString()).ToArray();
+
+            string [] payloadNames = methodEnv.MethodDecl.CSSignature.Skip(1).Where(a => a.IsGeneric).Select(a => methodEnv.GenericTypeMapping[a.SwiftTypeSpec.ToString()].PayloadName).ToArray();
+            string [] metadataNames = methodEnv.MethodDecl.CSSignature.Skip(1).Where(a => a.IsGeneric).Select(a => methodEnv.GenericTypeMapping[a.SwiftTypeSpec.ToString()].MetadataName).ToArray();
+            string [] typeNames = methodEnv.MethodDecl.CSSignature.Skip(1).Where(a => a.IsGeneric).Select(a => methodEnv.GenericTypeMapping[a.SwiftTypeSpec.ToString()].PlaceholderName).ToArray();
+            string [] argumentNames = methodEnv.MethodDecl.CSSignature.Skip(1).Where(a => a.IsGeneric).Select(a => a.Name).ToArray();
+
+            string swiftSelf = string.Empty;
+            if (methodEnv.ParentDecl is StructDecl structDecl && MarshallingHelpers.StructIsMarshalledAsCSStruct(structDecl))
+                swiftSelf = methodEnv.ParentDecl.Name;
 
-            var wrapperEmitter = new WrapperEmitter(methodEnv, signatureHandler);
-            wrapperEmitter.Emit(writer);
-            PInvokeEmitter.EmitPInvoke(writer, methodEnv, signatureHandler);
-            writer.WriteLine();
+            var voidReturn = methodEnv.MethodDecl.CSSignature.First().SwiftTypeSpec.IsEmptyTuple;
+            var hasReturn = !(methodEnv.RequiresIndirectResult || voidReturn);
+
+            csWriter.WriteLine($$"""
```

Comment:
This changes the approach from passing the code writer and emitting code on the way to recursively building a string from subnodes. I think this might be a valid approach to discuss further.